web/src/components/organisms/Project/ModelsMenu/hooks.ts

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { Model } from "@reearth-cms/components/molecules/Schema/types";
+import {
+  useGetModelsQuery,
+  useCreateModelMutation,
+  useCheckModelKeyAvailabilityLazyQuery,
+} from "@reearth-cms/gql/graphql-client-api";
+
+type Params = {
+  projectId?: string;
+  modelId?: string;
+};
+
+export default ({ projectId, modelId }: Params) => {
+  const [modelModalShown, setModelModalShown] = useState(false);
+  const [isKeyAvailable, setIsKeyAvailable] = useState(false);
+  const [CheckModelKeyAvailability, { data: keyData }] = useCheckModelKeyAvailabilityLazyQuery({
+    fetchPolicy: "no-cache",
+  });
+
+  const handleModelKeyCheck = useCallback(
+    async (projectId: string, key: string) => {
+      if (!projectId || !key) return false;
+      const response = await CheckModelKeyAvailability({ variables: { projectId, key } });
+      return response.data ? response.data.checkModelKeyAvailability.available : false;
+    },
+    [CheckModelKeyAvailability],
+  );
+
+  useEffect(() => {
+    setIsKeyAvailable(keyData?.checkModelKeyAvailability.available ?? false);
+  }, [keyData?.checkModelKeyAvailability]);
+
+  const { data } = useGetModelsQuery({
+    variables: { projectId: projectId ?? "", first: 100 },
+    skip: !projectId,
+  });
+
+  const models = useMemo(() => {
+    return (data?.models.nodes ?? [])
+      .map<Model | undefined>(model =>
+        model
+          ? {
+              id: model.id,
+              description: model.description,
+              name: model.name,
+              key: model.key,
+              schema: {
+                id: model.schema.id,
+                fields: model.schema.fields.map(field => ({
+                  id: field.id,
+                  description: field.description,
+                  title: field.title,
+                  type: field.type,
+                  key: field.key,
+                  unique: field.unique,
+                  required: field.required,
+                  typeProperty: field.typeProperty,
+                })),
+              },
+            }
+          : undefined,
+      )
+      .filter((model): model is Model => !!model);
+  }, [data?.models.nodes]);
+
+  const rawModel = useMemo(
+    () => data?.models.nodes.find((p: any) => p?.id === modelId),
```

Comment:
```suggestion
    () => data?.models.nodes.find((node: any) => node?.id === modelId),
```
Just a suggestion.
It is more readable this way, and adding a proper type would be a plus.

Reply:
@nourbalaha , done.
Could not type it though, It is from the generated GQL code, and all similar cases in re:earth are not typed also.